app/assets/javascripts/discourse/app/controllers/flag.js

```diff
@@ -60,15 +60,16 @@ export default Controller.extend(ModalFunctionality, {
     this._penalize("showSilenceModal", performAction);
   },
 
-  async _penalize(adminToolMethod, performAction) {
+  _penalize(adminToolMethod, performAction) {
     if (this.adminTools) {
-      let createdBy = await User.findByUsername(this.model.username);
-      let postId = this.model.id;
-      let postEdit = this.model.cooked;
-      return this.adminTools[adminToolMethod](createdBy, {
-        postId,
-        postEdit,
-        before: performAction,
+      User.findByUsername(this.model.username).then((createdBy) => {
```

Comment:
return it please :) it's always better, it allows other code to work with the promise if needed.